pkg/scheduler/nodezone/node_zone.go

```diff
@@ -19,30 +19,42 @@ package nodezone
 import (
 	"context"
 	"fmt"
+	"time"
 
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
 	corelisters "k8s.io/client-go/listers/core/v1"
 	"k8s.io/klog/v2"
 	"k8s.io/kubernetes/pkg/scheduler/framework"
 )
 
 // NodeZone is a plugin that checks node zone.
 type NodeZone struct {
-	cmLister corelisters.ConfigMapLister
+	handler    framework.Handle
+	cmLister   corelisters.ConfigMapLister
+	podLister  corelisters.PodLister
+	nodeLister corelisters.NodeLister
 }
 
 var _ framework.PreFilterPlugin = &NodeZone{}
 var _ framework.FilterPlugin = &NodeZone{}
+var _ framework.PermitPlugin = &NodeZone{}
+var _ framework.ReservePlugin = &NodeZone{}
 var _ framework.EnqueueExtensions = &NodeZone{}
 
 const (
-	ConfigMap framework.GVK = "ConfigMap"
-
 	// Name is the name of the plugin used in the plugin registry and configurations.
 	Name = "NodeZone"
 
+	// AvailableZones is the number failure domains over which we should spread.
+	AvailableZones = 3
```

Comment:
Why is it fixed?

Reply:
I think 3 is the minimum value to ensure HA on cloud.